gobblin-runtime/src/main/java/gobblin/runtime/util/ClustersNames.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.runtime.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * Allows conversion of URLs identifying a Hadoop cluster (e.g. resource manager url or
+ * a job tracker URL) to a human-readable name.
+ *
+ * <p>The class will automatically load a resource named {@link #URL_TO_NAME_MAP_RESOURCE_NAME} to
+ * get a default mapping. It expects this resource to be in the Java Properties file format. The
+ * name of the property is the cluster URL and the value is the human-readable name.
+ *
+ * <p><b>IMPORTANT:</b> Don't forget to escape colons ":" in the file as those may be interpreted
+ * as name/value separators.
+ */
+public class ClustersNames {
+
+  public static final String URL_TO_NAME_MAP_RESOURCE_NAME = "GobblinClustersNames.properties";
+  private static final Logger LOG = LoggerFactory.getLogger(ClustersNames.class);
+
+  private static ClustersNames THE_INSTANCE;
+
+  private Properties _urlToNameMap = new Properties();
```

Comment:
LinkedIn's coding conventions call for using "_" in instance variables. Personally, I dislike having to constantly type "this." to distinguish instance variables.